### PR TITLE
TER-103 external bucket drop role

### PR DESCRIPTION
**Pull request by martinjunger**
Jira: TER-103
driver-common: https://github.com/keboola/php-storage-driver-common/pull/34
Connection: https://github.com/keboola/connection/pull/4193

### CODE REVIEW

File: src/Handler/Bucket/Drop/RevokeBucketAccessFromReadOnlyRoleHandler.php [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    public function __invoke(
+        Message $credentials, // project credentials
+        Message $command,
+        array $features
+    ): ?Message {
+        assert($credentials instanceof GenericBackendCredentials);
+        assert($command instanceof RevokeBucketAccessFromReadOnlyRoleCommand);
+        assert(
+            $command->getProjectReadOnlyRoleName() !== '',
+            'GrantBucketAccessToReadOnlyRoleCommand.projectReadOnlyRoleName is required'
```
Comment (jirkasemmler) [inline src/Handler/Bucket/Drop/RevokeBucketAccessFromReadOnlyRoleHandler.php:38]: Revoke

File: tests/functional/UseCase/Bucket/GrantRevokeBucketAccessToReadOnlyRoleTest.php [hunk 2]
```diff
@@ -122,5 +125,31 @@ public function testWSCanAccessExternalBucketAfterGrantingToRO(): void
...
+            $wsConnection->fetchAllAssociative(
+                sprintf(
+                    'SELECT * FROM %s.%s',
+                    TeradataQuote::quoteSingleIdentifier($externalBucketDatabaseName),
+                    TeradataQuote::quoteSingleIdentifier('TESTTABLE')
+                )
+            );
+            $this->fail('Should fail. Bucket is no longer registered as external');
+        } catch (Throwable $e) {
+            $this->assertStringContainsString('The user does not have SELECT access to', $e->getMessage());
```
Comment (jirkasemmler) [inline tests/functional/UseCase/Bucket/GrantRevokeBucketAccessToReadOnlyRoleTest.php:152]: to je ta exception co hledas tady ? https://github.com/keboola/storage-api-php-client/pull/997#issuecomment-1424354103

**Review by jirkasemmler — CHANGES_REQUESTED** [review]

File: tests/functional/UseCase/Bucket/GrantRevokeBucketAccessToReadOnlyRoleTest.php (hunk 2 above)
Comment (martinjunger) [reply tests/functional/UseCase/Bucket/GrantRevokeBucketAccessToReadOnlyRoleTest.php:152]: jo, to bude nejspis ona

File: src/Handler/Bucket/Drop/RevokeBucketAccessFromReadOnlyRoleHandler.php (hunk 1 above)
Comment (martinjunger) [reply src/Handler/Bucket/Drop/RevokeBucketAccessFromReadOnlyRoleHandler.php:38]: fixnuto

**Review by jirkasemmler — APPROVED** [review]
cajk, jen composer

**Pull request merged**